File: Source/Typescript/src/app/model/Exam.ts
import { Vote } from "./Vote";

export interface Exam {
    examId:string;
    name: string;
    classId:string;
    subject: string;
    duration: number;
    provider: string;
    numberOfQuestion: number;
    difficultyLevel: string;
    createdDate: string;
    lastUpdatedDate: string;
    due: string;
    maxAttempts:number;
    status: string;
    type: string;
    id: string;
    new: boolean;
    votes:Vote[];
    upvoteCount:number;
    downvoteCount:number;
    commentCount:number;
}

File: Source/Java/AutoGraderService/src/main/java/com/nhson/autograderservice/exam/entities/Exam.java
package com.nhson.autograderservice.exam.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Entity
@Table(name = "exams")
@NoArgsConstructor
public class Exam {
    @Id
    private String examId;
    @Column(name = "exam_name")
    private String name;
    @Column(name = "class_id")
    private String classId;
    @Enumerated(EnumType.STRING)
    private Subject subject;
    private Integer duration;
    private String provider;
    @Column(name = "number_of_question")
    private Integer numberOfQuestion;
    @Enumerated(EnumType.STRING)
    private DifficultyLevel difficultyLevel;
    private LocalDateTime createdDate;
    private LocalDateTime lastUpdatedDate;
    private LocalDateTime due;
    private Integer maxAttempts;
    @Enumerated(EnumType.STRING)
    private Status status;
    @Enumerated(EnumType.STRING)
    private Type type;
}
